rust/lance-core/src/io/writer/statistics.rs

```diff
@@ -1994,4 +1912,248 @@ mod tests {
         assert_eq!(batch.schema().as_ref(), &expected_schema);
         assert_eq!(batch, expected_batch);
     }
+
+    // Property 1: for all values, if an array is entirely that value then it
+    // should become both the min and max. There are some exceptions:
+    // * NaN, null are ignored for min and max
+    // * string / binary more than 64 bytes are truncated.
+    fn assert_min_max_constant_property(
+        value: ScalarValue,
+        with_nulls: bool,
+    ) -> std::result::Result<(), TestCaseError> {
+        let array_scalar = value.to_scalar();
+        let (array, _) = array_scalar.get();
+        let array = make_array(array.to_data());
+
+        let array = if with_nulls {
+            let nulls = new_null_array(array.data_type(), 1);
+            interleave(&[&array, &nulls], &[(1, 0), (0, 0)]).unwrap()
+        } else {
+            array
+        };
+
+        let stats = collect_statistics(&[&array]);
+        prop_assert_eq!(
+            stats,
+            StatisticsRow {
+                null_count: if with_nulls { 1 } else { 0 },
+                min_value: value.clone(),
+                max_value: value,
+            },
+            "Statistics are wrong for input data: {:?}",
+            array
+        );
+        Ok(())
+    }
+
+    #[test]
+    fn test_min_max_constant_property() {
+        let values = [
+            ScalarValue::Boolean(Some(true)),
+            ScalarValue::Boolean(Some(false)),
+            ScalarValue::Int32(Some(1)),
```

Comment:
Maybe add some more int cases?
```suggestion
            ScalarValue::UInt8(Some(0)),
            ScalarValue::UInt16(Some(u16::MIN)),
            ScalarValue::UInt32(Some(1)),
            ScalarValue::UInt64(Some(u64::MAX)),
            ScalarValue::Int8(Some(0)),
            ScalarValue::Int16(Some(i16::MAX)),
            ScalarValue::Int32(Some(1)),
            ScalarValue::Int64(Some(i64::MIN)),
```

Reply:
Yeah I suppose I could add the remaining data types.